src/middlewared/middlewared/plugins/zfs_/dataset.py

```diff
@@ -194,7 +195,9 @@ def do_delete(self, id_, options):
         args = []
         if force:
             args += ['-f']
-        if recursive:
+        if options['recursively_remove_dependents']:
```

Comment:
In this same method you're assigning variables based on the options that are set but this one option you're not doing the same. Let's be consistent and either
1. remove the `recursive` and `force` variables and check `options["recursive"]` etc
2. OR pull the `recursively_remove_dependents` into its own variable as well.